bin/crew

```diff
@@ -1535,16 +1509,85 @@ def upload(pkg_name = nil, pkg_version = nil, gitlab_token = nil, gitlab_token_u
             puts "expected sha256 hash=#{new_sha256}"
             puts "upstream sha256 hash=#{upstream_sha256}"
           end
-          puts "#{output}. Checksum mismatch. Skipping binary_sha256 update in #{pkg_file}...".lightred
+          puts "#{output}. Checksum mismatch...".lightred
           next
         end
       else
         puts output.lightred
-        puts "#{output}. Unable to upload. Skipping binary_sha256 update in #{pkg_file}...".lightred
+        puts "#{output}. Unable to upload...".lightred
         next
       end
     end
 
+    # Upload python wheels if we are dealing with a pip package, but only
+    # if a gitlab token username is set. (The generic repo does not
+    # require a gitlab token username.)
+    if @pkg.superclass.to_s == 'Pip'
+      pip_config = `pip config list`.chomp
+      Kernel.system 'pip config --user set global.trusted-host gitlab.com', %i[err out] => File::NULL unless pip_config.include?("global.trusted-host='gitlab.com'")
+
+      pip_cache_dir = `pip cache dir`.chomp
+      wheels = `find #{pip_cache_dir} -type f -name \"*.whl\" -print`.chomp.split
+      unless wheels.empty?
+        wheels.each do |wheel|
+          puts "Uploading #{wheel}.\nNote that a '400 Bad Request' error here means the wheel has already been uploaded.".orange
+          # Note that this uses the python twine from https://github.com/pypa/twine/pull/1123
+          abort 'Twine is broken, cannot upload python wheels.'.lightred unless system('twine --help', %i[out err] => File::NULL)
+          system("twine upload -u #{gitlab_token_username} -p #{gitlab_token} --repository-url #{CREW_GITLAB_PKG_REPO}/pypi --non-interactive #{wheel}", %i[err] => File::NULL)
+          FileUtils.rm_f wheel
+        end
+      end
+    end
+    puts "\e[1A\e[K🎉 Uploads complete for #{package}. 🎉\r\n".lightgreen
+    puts "\e[1A\e[K🎉 Updating package file for #{package}. 🎉\r\n".lightgreen
+    update_package_binary_hashes(pkg_name, pkg_version, binary_compression)
+  end
+end
+
+def update_package_binary_hashes(pkg_name = nil, pkg_version = nil, binary_compression = nil)
```

Comment:
This feels much better suited as a script in tools, given that it's quite developer specific, and beyond that pretty CI specific.